llvm/lib/Target/AArch64/GISel/AArch64O0PreLegalizerCombiner.cpp

```diff
@@ -165,6 +165,8 @@ bool AArch64O0PreLegalizerCombiner::runOnMachineFunction(MachineFunction &MF) {
   CombinerInfo CInfo(/*AllowIllegalOps*/ true, /*ShouldLegalizeIllegal*/ false,
                      /*LegalizerInfo*/ nullptr, /*EnableOpt*/ false,
                      F.hasOptSize(), F.hasMinSize());
+  CInfo.MaxIterations = 1;
```

Comment:
Please add a comment, perhaps calling back to this PR # so readers will know why we set it to 1.